x-pack/plugins/cloud_security_posture/public/components/csp_page_template.tsx

```diff
@@ -0,0 +1,175 @@
+
+interface QueryError {
+  body: {
+    error: string;
+    message: string;
+    statusCode: number;
+  };
+  name: string;
+}
+
+const activeItemStyle = { fontWeight: 700 };
+
+export const getSideNavItems = (
+  navigationItems: Record<string, CspNavigationItem>
+): NonNullable<KibanaPageTemplateProps['solutionNav']>['items'] =>
+  Object.entries(navigationItems)
+    .filter(([_, navigationItem]) => !navigationItem.disabled)
+    .map(([id, navigationItem]) => ({
+      id,
+      name: navigationItem.name,
+      renderItem: () => (
+        <NavLink to={navigationItem.path} activeStyle={activeItemStyle}>
+          {navigationItem.name}
+        </NavLink>
+      ),
+    }));
+
+const DEFAULT_PAGE_PROPS: KibanaPageTemplateProps = {
+  solutionNav: {
+    name: CLOUD_SECURITY_POSTURE,
+    items: getSideNavItems({
+      dashboard: allNavigationItems.dashboard,
+      findings: allNavigationItems.findings,
+      benchmark: allNavigationItems.benchmarks,
+    }),
+  },
+  restrictWidth: false,
+};
+
+export const DEFAULT_NO_DATA_CONFIG: KibanaPageTemplateProps['noDataConfig'] = {
+  pageTitle: DEFAULT_NO_DATA_TEXT.PAGE_TITLE,
+  solution: DEFAULT_NO_DATA_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {},
+};
+
+const getPackageNotInstalledConfig = (
+  cisIntegrationLink: string
+): KibanaPageTemplateProps['noDataConfig'] => ({
+  pageTitle: PACKAGE_NOT_INSTALLED_TEXT.PAGE_TITLE,
+  solution: PACKAGE_NOT_INSTALLED_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {
+    elasticAgent: {
+      href: cisIntegrationLink,
+      title: PACKAGE_NOT_INSTALLED_TEXT.BUTTON_TITLE,
+      description: PACKAGE_NOT_INSTALLED_TEXT.DESCRIPTION,
+    },
+  },
+});
+
+const DefaultLoading = () => <CspLoadingState>{LOADING}</CspLoadingState>;
+
+const DefaultError = (error: QueryError) => (
+  <EuiEmptyPrompt
+    color="danger"
+    iconType="alert"
+    title={
+      <>
+        <EuiTitle>
+          <h2>
+            <FormattedMessage
+              id="xpack.csp.pageTemplate.loadErrorMessage"
+              defaultMessage="We couldn't fetch your cloud security posture data"
+            />
+          </h2>
+        </EuiTitle>
+        {error.body?.error && error.body?.statusCode && (
+          <EuiTitle size="xs">
+            <h5>{`${error.body.error} (${error.body.statusCode})`}</h5>
+          </EuiTitle>
+        )}
+        {error.body?.message && (
+          <EuiTitle size="xs">
+            <h5>{error.body.message}</h5>
+          </EuiTitle>
+        )}
+      </>
+    }
+  />
+);
+
+export const CspPageTemplate: React.FC<
+  KibanaPageTemplateProps & {
+    loadingRender?: () => JSX.Element;
+    errorRender?: (error: QueryError) => JSX.Element;
```

Comment:
what you described is what happens so no worries. if you pass `query` you will get the `error` into `errorRender` but if you don't pass a query, obviously, the `CspPageTemplate` can't know when to display the error state, so in case you didn't pass the query, you are expected to just handle the entire rendering by yourself as we did before for some pages. honestly, this component should be enough 95% of the time, but the option to handle it yourself is there if the case arises.